experimental/golden_response_checker.go

```diff
@@ -0,0 +1,122 @@
+package experimental
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// CheckGoldenDataResponse will verify that the stored file matches the given data.DataResponse
+// when the updateFile flag is set, this will both add errors to the response and update the saved file
+func CheckGoldenDataResponse(path string, dr *backend.DataResponse, updateFile bool) error {
+	saved, err := readGoldenFile(path)
+	if err != nil {
+		err = fmt.Errorf("error reading golden file:  %s\n%s", path, err.Error())
+	} else {
+		if diff := cmp.Diff(saved.Error, dr.Error); diff != "" {
+			err = fmt.Errorf("errors mismatch %s (-want +got):\n%s", path, diff)
+		}
+		if diff := cmp.Diff(len(saved.Frames), len(dr.Frames)); diff != "" {
+			err = fmt.Errorf("Frame count mismatch (-want +got):\n%s", diff)
```

Comment:
the previous error would be masked by this error if both conditions are true. Consider returning, or making an array of errors, and wrapping them all together.